### ISSUE TITLE
Implement AsRef<Path> for Arc<PathBuf>

### ISSUE THREAD

**Issue by vitiral** [rust-lang/rust#58469]
Possibly also implement `Borrow` and `Deref`.

I would like to use `Arc<PathBuf>` to be the equivalent of `PathBuf` for traits in my library, `path_abs`.

However, I get errors like this:

```
751 | impl PathMut for Arc<PathBuf> {
    |      ^^^^^^^ the trait `std::convert::AsRef<std::path::Path>` is not implemented for `alloc::sync::Arc<std::path::PathBuf>`
    |
    = help: the following implementations were found:
              <alloc::sync::Arc<T> as std::convert::AsRef<T>>
    = note: required because of the requirements on the impl of `PathInfo` for `alloc::sync::Arc<std::path::PathBuf>`
```

**Comment by abonander** [comment]
You can also get `Arc<Path>` via `.into()` on `&Path` or `PathBuf`.

**Comment by vitiral** [comment]
Of course, but that doesn't let you generalize for trait bounds in a library.

**Comment by JanBeh** [comment]
Would be solved by #45742 (which is difficult), so maybe this can be closed?

**Comment by Enselic** [comment]
Triage: There were no objections to close as duplicate, so let's do that.

Duplicate of #45742
